src/test/kotlin/con/exactpro/th2/codec/qfj/QFJCodecTest.kt

```diff
@@ -0,0 +1,431 @@
+package con.exactpro.th2.codec.qfj
+
+import com.exactpro.th2.codec.qfj.QFJCodec
+import com.exactpro.th2.codec.qfj.QFJCodecSettings
+import com.exactpro.th2.common.grpc.AnyMessage
+import com.exactpro.th2.common.grpc.ConnectionID
+import com.exactpro.th2.common.grpc.Direction
+import com.exactpro.th2.common.grpc.EventID
+import com.exactpro.th2.common.grpc.ListValue
+import com.exactpro.th2.common.grpc.Message
+import com.exactpro.th2.common.grpc.MessageGroup
+import com.exactpro.th2.common.grpc.MessageID
+import com.exactpro.th2.common.grpc.MessageMetadata
+import com.exactpro.th2.common.grpc.RawMessage
+import com.exactpro.th2.common.grpc.RawMessageMetadata
+import com.exactpro.th2.common.grpc.Value
+import com.google.protobuf.ByteString
+import com.google.protobuf.Timestamp
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import quickfix.ConfigError
+import quickfix.DataDictionary
+import quickfix.FieldException
+import quickfix.Group
+import quickfix.field.ApplID
+import quickfix.field.BeginString
+import quickfix.field.HopCompID
+import quickfix.field.MsgType
+import quickfix.field.OnBehalfOfCompID
+import quickfix.field.PartyID
+import quickfix.field.PartyIDSource
+import quickfix.field.PartyRole
+import quickfix.field.SenderCompID
+import quickfix.field.Side
+import quickfix.field.SignatureLength
+import quickfix.field.TargetCompID
+import quickfix.field.TestReqID
+import java.time.Instant
+import java.util.TreeMap
+import quickfix.field.NoHops
+import quickfix.field.NoPartyIDs
+import quickfix.field.NoSides
+import quickfix.field.Signature
+
+class QFJCodecTest {
+
+    companion object {
+        private var codec: QFJCodec? = null
+        private var messageGroup: MessageGroup? = null
+        private var messageGroupNoHeader: MessageGroup? = null
+        private var rawMessageGroup: MessageGroup? = null
+        private var strFixMessage: String? = null
+        private var timestampSeconds = Instant.now().epochSecond
+        private var timestampNano = Instant.now().nano
+        private var checksumValue: String = ""
+
+        private fun getBodyLength(message: String?): String {
+            return message!!.substring(
+                message.indexOf("\u00019=") + 3,
+                message.indexOf("\u0001", message.indexOf("\u00019=") + 1)
+            )
```

Comment:
Can be replaced with substringAfter/stringBefore calls. Less efficient, but won't be critical in test. Or we can create a generic method which extracts tag value from a string